.github/workflows/continuous-integration-workflow.yml

```diff
@@ -51,16 +51,12 @@ jobs:
           push: true
           tags: ${{ secrets.DOCKERHUB_USERNAME }}/${{ secrets.DOCKERHUB_REPOSITORY }}:latest
   job-three:
-    name: Deploy Heroku
+    name: Run Azure Login with OpenID Connect
     runs-on: ubuntu-latest
     needs: [job-one, job-two]
+    permissions:
+      id-token: write
+      contents: read
     steps:
-      - name: Check out repository
-        uses: actions/checkout@v2
-      - name: Deploy to Heroku
-        uses: akhileshns/heroku-deploy@v3.12.12
-        with:
-          heroku_api_key: ${{ secrets.HEROKU_API_KEY }}
-          heroku_app_name: ${{ secrets.HEROKU_APP_NAME }}
-          heroku_email: ${{ secrets.HEROKU_EMAIL }}
-          usedocker: true
\ No newline at end of file
+    - name: 'Run Azure CLI commands'
+      run: curl -dH -X POST ${{ secrets.WEBHOOK_AZ }}
```

Comment:
One nice addition here can be adding the [--fail](https://curl.se/docs/manpage.html#-f) flag, as otherwise curl reports success as long as a successful connection was made, even if the response reports a failure (e.g. a status code > 400) which can lead to confusing pipeline results